metric.go

```diff
@@ -113,16 +113,51 @@ type MetricDataArray []*MetricData
 
 // for ES
 type MetricDefinition struct {
-	Id         string   `json:"id"`
-	OrgId      int      `json:"org_id"`
-	Name       string   `json:"name" elastic:"type:string,index:not_analyzed"` // graphite format
-	Metric     string   `json:"metric"`                                        // kairosdb format (like graphite, but not including some tags)
-	Interval   int      `json:"interval"`                                      // minimum 10
-	Unit       string   `json:"unit"`
-	Mtype      string   `json:"mtype"`
-	Tags       []string `json:"tags" elastic:"type:string,index:not_analyzed"`
-	LastUpdate int64    `json:"lastUpdate"` // unix timestamp
-	Partition  int32    `json:"partition"`
+	Id           string   `json:"id"`
+	OrgId        int      `json:"org_id"`
+	Name         string   `json:"name" elastic:"type:string,index:not_analyzed"` // graphite format
+	Metric       string   `json:"metric"`                                        // kairosdb format (like graphite, but not including some tags)
+	Interval     int      `json:"interval"`                                      // minimum 10
+	Unit         string   `json:"unit"`
+	Mtype        string   `json:"mtype"`
+	Tags         []string `json:"tags" elastic:"type:string,index:not_analyzed"`
+	LastUpdate   int64    `json:"lastUpdate"` // unix timestamp
+	Partition    int32    `json:"partition"`
+	NameWithTags string   `json:"nameWithTags"`
+}
+
+func NewMetricDefinition(name string, tags []string) *MetricDefinition {
```

Comment:
not sure how useful this method is, since everywhere we create a MetricDefinition, we want to set all other fields as well. If i were you I would check in all places where MetricDefinitions are created and see which fields they initialize, maybe you can just add args to this func for them.

Reply:
true, but then the signature would end up being really long with a lot of arguments. probably it's more tidy to just keep instantiating manually and remove that method again